tests/distributed/_test_distributed.py

```diff
@@ -0,0 +1,156 @@
+import copy
+import subprocess
+import sys
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict, List
+
+import numpy as np
+from sklearn.datasets import make_blobs, make_regression
+from sklearn.metrics import accuracy_score
+
+import lightgbm as lgb
```

Comment:
The compilation step takes the most time and once that's done using the `--precompile` flag makes the installation very fast. I thought we may use the python package later for loading the booster and checking some stuff like what's currently done for the categoricals here: https://github.com/microsoft/LightGBM/blob/0a172d9e3b292110d90878ba7f0932623b80cbe2/tests/python_package_test/test_dask.py#L315-L323 for example.

One thing that does take a bit more time is this: https://github.com/microsoft/LightGBM/blob/0a172d9e3b292110d90878ba7f0932623b80cbe2/.ci/test.sh#L107-L108 I placed these tests below that because they need numpy and scikit-learn but could maybe move this a bit up to just install those and avoid all the other stuff that gets installed there. What do you think?

Reply:
oooo ok yeah, that makes sense. I wasn't thinking about how quick it is with `--precompile`.

I'd definitely support moving these tests above that line and just `conda install`-ing what you need inside the task's code in `test.sh`.

We do the same thing for other tasks like `check-docs` and `if-else`.

https://github.com/microsoft/LightGBM/blob/0a172d9e3b292110d90878ba7f0932623b80cbe2/.ci/test.sh#L31

https://github.com/microsoft/LightGBM/blob/0a172d9e3b292110d90878ba7f0932623b80cbe2/.ci/test.sh#L80

Reply:
Addressed in 9f51ad8

Reply:
excellent, thanks!